Configuration.md

```diff
@@ -0,0 +1,198 @@
+scrape_configs:  
+  - job_name:  'prometheus'  
+    static_configs:
+         - targets: ['localhost:9090']
+
+  # metrics_path defaults to '/metrics'
+```
+
+```yml
+ # docker-compose.yml
+  command:
+       - "--config.file=/etc/prometheus/prometheus.yml"  # chỉ định configure file vào đường dẫn
+```
+
+---
+
+Prometheus hỗ trợ 2 kiểu rules : Recording rule và Alerting rule. Các statement được chứa trong tệp YAML `rule_files` và được Prometheus load  vào.
+
+*_Recording rules_*
+
+Cho phép ta tính toán trước các biểu thức cần thiết hoặc tính toán đắt tiền và lưu kết quả của chúng dưới dạng một time series mới.
+
+Truy vấn kết quả được tính toán trước thường sẽ nhanh hơn nhiều so với thực hiện biểu thức gốc mỗi khi cần. Điều này đặc biệt hữu ích cho bảng điều khiển, cần truy vấn cùng một biểu thức liên tục mỗi lần chúng làm mới.
+
+```yml
+#rule_files
+groups:
+  - name: example # tên của group
+    rules:
+    #Tên của time series mới để lưu kết quả  
+    - record: node_memory_MemFree_percent
+      # biểu thức tính metric
+      expr: 100 - (100 * node_memory_MemFree_bytes / node_memory_MemTotal_bytes)
+
+
+```
+
+Check rules sử dụng promtool:
+
+![ ](https://github.com/quynhvuongg/Picture/blob/master/prometheus5.png?raw=true)
+  
+```yml
+#prometheus.yml
+rule_files:  
+- "rule_files"
+```
+
+*_Alerting rules_*
+
+Cho phép ta xác định các điều kiện cảnh báo dựa trên biểu thức Prometheus và gửi thông báo kích hoạt cảnh báo tới dịch vụ bên ngoài.
+
+```yml
+#rule_files
+groups:
+- name: example
+  rules:
+# Cảnh báo cho bất kì instance không tới được trong 2m
+  - alert: service_down #tên của alert
+    expr: up == 0 #biểu thức đánh giá
+    for: 2m #thời gian chờ xử lý trước khi kích hoạt cảnh báo
+    #label được đính kèm cùng alert ,nhãn đã tồn tại nào xung đột sẽ bị ghi đè
+    labels:
+      severity: page
+    annotations:
+      summary: "Instance {{ $labels.instance }} down"
+      description: "{{ $labels.instance }} of job {{ $labels.job }} has been down for more than 2 minutes."
+#annotations chỉ định một bộ nhãn thông tin có thể được sử dụng để lưu trữ thông tin bổ sung dài hơn như mô tả cảnh báo hoặc liên kết runbook.
+```
+
+```yml
+#prometheus.yml
+rule_files:  
+- "rule_files"
+```
+
+```yml
+#docker-compose.yml
+ volumes:  
+   - ...
+   - ./rule_files:/etc/prometheus/rule_files
+```
+
+![ ](https://github.com/quynhvuongg/Picture/blob/master/prometheus6.png?raw=true)
+
+*_Inspecting alerts during runtime_*
+
+Để kiểm tra thủ công cảnh báo nào đang hoạt động (pending or firing), chuyển đến tab "Alerts" trong Prometheus. Nó sẽ cho ta thấy các bộ nhãn chính xác mà mỗi cảnh báo được xác định hiện đang hoạt động.
+
+Đối với cảnh báo pending and firing, Prometheus cũng lưu trữ time series tổng hợp có dạng: `ALERTS{alertname="<alert name>", alertstate="pending|firing", <additional alert labels>}`.
+
+![ ](https://github.com/quynhvuongg/Picture/blob/master/Alert2.png?raw=true )
+
+*_Sending alert notifications_*
```

Comment:
tương tự, bôi đậm, in nghiêng thế này dễ xót mà ko nhấn mạnh được

https://www.markdownguide.org/basic-syntax/#headings